src/sentry/static/sentry/app/views/settings/incidentRules/triggers/actionsPanel/index.tsx

```diff
@@ -55,17 +87,37 @@ class ActionsPanel extends React.PureComponent<Props> {
     onChange(replaceAtArrayIndex(actions, index, newAction));
   }
 
-  getFullActionTitle({
-    type,
-    integrationName,
-  }: Pick<MetricAction, 'type'> & {
-    integrationName?: MetricAction['integrationName'] | null;
-  }) {
-    return `${ActionLabel[type]}${integrationName ? ` - ${integrationName}` : ''}`;
-  }
+  handleAddAction = (value: {label: string; value: string}) => {
+    const {availableActions} = this.props;
+
+    const actionConfig =
+      availableActions &&
+      availableActions.find(
+        availableAction => this.getActionUniqueKey(availableAction) === value.value
+      );
 
-  handleAddAction = (value: {label: string; value: Action['type']}) => {
-    this.props.onAdd(value.value);
+    if (!actionConfig) {
+      addErrorMessage(t('There was a problem adding an action'));
+      Sentry.setExtras({
+        integrationId: value,
+      });
+      Sentry.captureException(new Error('Unable to add an action'));
+      return;
+    }
+
+    const action: Action = {
```

Comment:
Changed to return an action instead of just the selected item so that we can colocate the unique key logic in the same component.